### PR TITLE
Ship SARIF template as a JSON file in external data folder

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Storing the template in a `.json` file makes it easier to read, esp. because of proper indentation, and also to format them in case of modification.

Inlining the JSON as a string in the source code makes both of these difficult.

WDYT?

### CODE REVIEW

**Comment by MichaelChirico** [comment]
cc @shaopeng-gh who authored SARIF support.

looks ok to me. is there any prior art on what templated json should look like?

File: inst/extdata/sarif-template.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      ],
+      "columnKind": "utf16CodeUnits",
+      "originalUriBaseIds": {
+        "ROOTPATH": {
+          "uri": "file:///C:/repos/repototest/"
+        }
+      }
+    }
+  ]
+}
```
Comment (AshesITR) [inline inst/extdata/sarif-template.json:58]: Missing terminal newline

Comment (IndrajeetPatil) [reply inst/extdata/sarif-template.json]: Thanks. Fixed. 

VS code formatter wouldn't let me add it. So had to do it from RStudio.

**Review by AshesITR — APPROVED** [review]
This should be safe regardless of feedback since all we do is move the constant to another file.

**Pull request merged**